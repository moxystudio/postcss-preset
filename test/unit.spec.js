'use strict';

const preset = require('..');

jest.mock('postcss-import', () => (options) => ['postcss-import', options].filter((val) => val));
jest.mock('postcss-url', () => (options) => ['postcss-url', options].filter((val) => val));
jest.mock('postcss-advanced-variables', () => (options) => ['postcss-advanced-variables', options].filter((val) => val));
jest.mock('postcss-mixins', () => (options) => ['postcss-mixins', options].filter((val) => val));
jest.mock('postcss-calc', () => (options) => ['postcss-calc', options].filter((val) => val));
jest.mock('postcss-preset-env', () => (options) => ['postcss-preset-env', options].filter((val) => val));
jest.mock('postcss-color-function', () => (options) => ['postcss-color-function', options].filter((val) => val));

afterEach(jest.resetAllMocks);

it('should have a good base config', () => {
    expect(preset()).toMatchSnapshot();
});

it('should pass options.import to postcss-import ', () => {
    expect(preset({ import: 'foo' })).toMatchSnapshot();
});

it('should pass options.mixins to postcss-mixins ', () => {
    expect(preset({ mixins: 'foo' })).toMatchSnapshot();
});

it('should pass options.advancedVariables to postcss-advanced-variables ', () => {
    expect(preset({ advancedVariables: 'foo' })).toMatchSnapshot();
});

it('should disable postcss-url if options.url is false', () => {
    expect(preset({ url: false })).toMatchSnapshot();
});

it('should pass options.url to postcss-url', () => {
    expect(preset({ url: { foo: 'bar' } })).toMatchSnapshot();
});

it('should pass options.browsers to postcss-preset-env ', () => {
    expect(preset({ browsers: 'foo' })).toMatchSnapshot();
});

it('should disable postcss-css-variables if options.cssVariables is false', () => {
    expect(preset({ cssVariables: false })).toMatchSnapshot();
});

it('should pass options.cssVariables to postcss-css-variables', () => {
    expect(preset({ cssVariables: { preserve: true } })).toMatchSnapshot();
});
