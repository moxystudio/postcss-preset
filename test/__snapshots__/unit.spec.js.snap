// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should enable postcss-url if options.url is true 1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
    ],
    Array [
      "postcss-url",
    ],
    Array [
      "postcss-mixins",
    ],
    Array [
      "postcss-conditionals",
    ],
    Array [
      "postcss-preset-env",
      Object {
        "autoprefixer": Object {
          "overrideBrowserslist": Array [
            "extends browserslist-config-google",
          ],
          "remove": false,
        },
        "browsers": Array [
          "extends browserslist-config-google",
        ],
        "features": Object {
          "custom-media-queries": true,
          "custom-properties": false,
          "nesting-rules": true,
        },
        "insertAfter": Object {
          "nesting-rules": [Function],
        },
        "preserve": false,
        "stage": 3,
      },
    ],
    Array [
      "postcss-color-function",
    ],
  ],
}
`;

exports[`should have a good base config 1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
    ],
    Array [
      "postcss-url",
      Object {
        "url": "rebase",
      },
    ],
    Array [
      "postcss-mixins",
    ],
    Array [
      "postcss-conditionals",
    ],
    Array [
      "postcss-preset-env",
      Object {
        "autoprefixer": Object {
          "overrideBrowserslist": Array [
            "extends browserslist-config-google",
          ],
          "remove": false,
        },
        "browsers": Array [
          "extends browserslist-config-google",
        ],
        "features": Object {
          "custom-media-queries": true,
          "custom-properties": false,
          "nesting-rules": true,
        },
        "insertAfter": Object {
          "nesting-rules": [Function],
        },
        "preserve": false,
        "stage": 3,
      },
    ],
    Array [
      "postcss-color-function",
    ],
  ],
}
`;

exports[`should pass options.browsers to postcss-cssnext  1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
    ],
    Array [
      "postcss-url",
      Object {
        "url": "rebase",
      },
    ],
    Array [
      "postcss-mixins",
    ],
    Array [
      "postcss-conditionals",
    ],
    Array [
      "postcss-preset-env",
      Object {
        "autoprefixer": Object {
          "overrideBrowserslist": "foo",
          "remove": false,
        },
        "browsers": "foo",
        "features": Object {
          "custom-media-queries": true,
          "custom-properties": false,
          "nesting-rules": true,
        },
        "insertAfter": Object {
          "nesting-rules": [Function],
        },
        "preserve": false,
        "stage": 3,
      },
    ],
    Array [
      "postcss-color-function",
    ],
  ],
}
`;

exports[`should pass options.import to postcss-import  1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
      "foo",
    ],
    Array [
      "postcss-url",
      Object {
        "url": "rebase",
      },
    ],
    Array [
      "postcss-mixins",
    ],
    Array [
      "postcss-conditionals",
    ],
    Array [
      "postcss-preset-env",
      Object {
        "autoprefixer": Object {
          "overrideBrowserslist": Array [
            "extends browserslist-config-google",
          ],
          "remove": false,
        },
        "browsers": Array [
          "extends browserslist-config-google",
        ],
        "features": Object {
          "custom-media-queries": true,
          "custom-properties": false,
          "nesting-rules": true,
        },
        "insertAfter": Object {
          "nesting-rules": [Function],
        },
        "preserve": false,
        "stage": 3,
      },
    ],
    Array [
      "postcss-color-function",
    ],
  ],
}
`;

exports[`should pass options.mixins to postcss-mixins  1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
    ],
    Array [
      "postcss-url",
      Object {
        "url": "rebase",
      },
    ],
    Array [
      "postcss-mixins",
      "foo",
    ],
    Array [
      "postcss-conditionals",
    ],
    Array [
      "postcss-preset-env",
      Object {
        "autoprefixer": Object {
          "overrideBrowserslist": Array [
            "extends browserslist-config-google",
          ],
          "remove": false,
        },
        "browsers": Array [
          "extends browserslist-config-google",
        ],
        "features": Object {
          "custom-media-queries": true,
          "custom-properties": false,
          "nesting-rules": true,
        },
        "insertAfter": Object {
          "nesting-rules": [Function],
        },
        "preserve": false,
        "stage": 3,
      },
    ],
    Array [
      "postcss-color-function",
    ],
  ],
}
`;

exports[`should pass options.url to postcss-url 1`] = `
Object {
  "plugins": Array [
    Array [
      "postcss-import",
    ],
    Array [
      "postcss-url",
      Object {
        "foo": "bar",
      },
    ],
    Array [
      "postcss-mixins",
    ],
    Array [
      "postcss-conditionals",
    ],
    Array [
      "postcss-preset-env",
      Object {
        "autoprefixer": Object {
          "overrideBrowserslist": Array [
            "extends browserslist-config-google",
          ],
          "remove": false,
        },
        "browsers": Array [
          "extends browserslist-config-google",
        ],
        "features": Object {
          "custom-media-queries": true,
          "custom-properties": false,
          "nesting-rules": true,
        },
        "insertAfter": Object {
          "nesting-rules": [Function],
        },
        "preserve": false,
        "stage": 3,
      },
    ],
    Array [
      "postcss-color-function",
    ],
  ],
}
`;
